Depending on packages which use `TextSelectionOverlay.fadeDuration` results in build failure

<!-- Thank you for using Flutter!

     If you are looking for support, please check out our documentation
     or consider asking a question on Stack Overflow:
      * https://flutter.dev/
      * https://api.flutter.dev/
      * https://stackoverflow.com/questions/tagged/flutter?sort=frequent

     If you have found a bug or if our documentation doesn't have an answer
     to what you're looking for, then fill out the template below. Please read
     our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports
-->

## Steps to Reproduce

If some app is depend on a library (ex. [flutter_math_fork](https://pub.dev/packages/flutter_math_fork)) that uses `TextSelectionOverlay.fadeDuration`, the app will not be build with flutter 2.12.0-4.1.pre or higher.

ref https://github.com/flutter/flutter/pull/98153

---

In the library [flutter_math_fork](https://pub.dev/packages/flutter_math_fork), `TextSelectionOverlay.fadeDuration` is used in the following line. 
https://github.com/simpleclub-extended/flutter_math_fork/blob/a6f0c766a660b38c767fb12486672165630a8a69/lib/src/widgets/selection/handle_overlay.dart#L60 

So, applications using flutter_math_fork cannot upgrade flutter until flutter_math_fork is updated.
On the other hand, if another library updates flutter to use `SelectionOverlay.fadeDuration`, the developer will need to modify that library's version. flutter_math_fork seems to be a popular library, but perhaps other libraries will have this problem as well. 
[search result](https://github.com/search?l=Dart&q=TextSelectionOverlay+fadeDuration&type=Code)

Would it be possible to add the following deprecated const value?

```dart
class TextSelectionOverlay {

  /// Controls the fade-in and fade-out animations for the toolbar and handles.
  @Deprecated(
      'Use `SelectionOverlay.fadeDuration` instead. '
      'This value was deprecated in Flutter 2.12.0-4.1.pre'
  )
  static const Duration fadeDuration = SelectionOverlay.fadeDuration;
}
```

**Expected results:** <!-- what did you want to see? -->

No build errors.

**Actual results:** <!-- what did you see? -->

An error occurs during build.

```
../../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40:
Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
```

<details>
<summary>Code sample</summary>

<!--
      Please create a minimal reproducible sample that shows the problem
      and attach it below between the lines with the backticks.

      To create it you can use `flutter create bug` command and update the `main.dart` file.

      Alternatively, you can use https://dartpad.dev/
      which is capable of creating and running small Flutter apps.

      Without this we will unlikely be able to progress on the issue, and because of that
      we regretfully will have to close it.
-->

```dart
```

</details>

<details>
  <summary>Logs</summary>

<!--
      Run your application with `flutter run --verbose` and attach all the
      log output below between the lines with the backticks. If there is an
      exception, please see if the error message includes enough information
      to explain how to solve the issue.
-->

```
../../../.pub-cache/hosted/[pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40](http://pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40):
Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
```

<!--
     Run `flutter analyze` and attach any output of that command below.
     If there are any analysis errors, try resolving them before filing this issue.
-->

```
```

<!-- Finally, paste the output of running `flutter doctor -v` here. -->

```
$ flutter doctor -v
[✓] Flutter (Channel beta, 2.12.0-4.1.pre, on macOS 12.3 21E230 darwin-arm,
    locale ja-JP)
    • Flutter version 2.12.0-4.1.pre at /Users/koji/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 680962aa75 (30 hours ago), 2022-03-16 13:46:03 -0700
    • Engine revision e9f57b5d0f
    • Dart version 2.17.0 (build 2.17.0-182.1.beta)
    • DevTools version 2.11.1

[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/koji/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 65.2.4
    • Dart plugin version 213.7227

[✓] VS Code (version 1.65.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>


<!--
      Consider also attaching screenshots and/or videos to better
      illustrate the issue.

      You can upload them directly on GitHub.
      Beware that video file size is limited to 10MB.
-->

Hi @maheshmnj, Thanks for checking.


the reproducible steps are following.

1. Change to stable channel ('flutter channel stable`)
2. Create new project
3. Add `flutter_math_fork`
4. Run application -> OK
5. Change to beta channel (`flutter channel beta`)
6. Run application -> NG

---

I created sample project.
https://github.com/koji-1009/TextSelectionOverlay_error

When the flutter channel is `stable`, the project is OK. However, if `beta' is selected, the application cannot be built.

<details>
  <summary>Logs</summary>

```
Launching lib/main.dart on sdk gphone64 arm64 in debug mode...
Running Gradle task 'assembleDebug'...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selectable.dart:459:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback((_) {
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay_manager.dart:138:28: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
      if (SchedulerBinding.instance!.schedulerPhase ==
                           ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay_manager.dart:140:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
        SchedulerBinding.instance!
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/web_selection_manager.dart:119:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:141:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
    if (SchedulerBinding.instance!.schedulerPhase ==
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:143:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback(_markNeedsBuild);
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:193:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
    if (SchedulerBinding.instance!.schedulerPhase ==
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:195:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback(_markNeedsBuild);
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40: Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
                                       ^^^^^^^^^^^^


FAILURE: Build failed with an exception.

* Where:
Script '/Users/koji/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 1148

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/koji/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 10s
Exception: Gradle task assembleDebug failed with exit code 1
```

</details>

Thanks

Thanks for the reproducible steps @koji-1009, I am able to reproduce the issue the app runs fine on `stable 2.10.3`  and fails to build on beta and master channel 

@chunhtai could this PR https://github.com/flutter/flutter/pull/98153 be the reason why the build fails?

<details>
<summary>logs</summary>

```
mahesh@Maheshs-MacBook-Air-M1 test_app % flutterm run
Could not load custom device from config index 0: Expected enabled to be a boolean.
Running "flutter pub get" in test_app...                            3.7s
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selectable.dart:459:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback((_) {
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay_manager.dart:138:28: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
      if (SchedulerBinding.instance!.schedulerPhase ==
                           ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay_manager.dart:140:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
        SchedulerBinding.instance!
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/web_selection_manager.dart:119:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:141:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
    if (SchedulerBinding.instance!.schedulerPhase ==
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:143:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback(_markNeedsBuild);
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:193:26: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
    if (SchedulerBinding.instance!.schedulerPhase ==
                         ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/overlay.dart:195:24: Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter_master/packages/flutter/lib/src/scheduler/binding.dart').
      SchedulerBinding.instance!.addPostFrameCallback(_markNeedsBuild);
                       ^
../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40: Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
                                       ^^^^^^^^^^^^


FAILURE: Build failed with an exception.

* Where:
Script '/Users/mahesh/Documents/flutter_master/packages/flutter_tools/gradle/flutter.gradle' line: 1156

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/mahesh/Documents/flutter_master/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
Running Gradle task 'assembleDebug'...   
```
</details> 


<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.3 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (20 hours ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.5:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.103, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
    • Flutter version 2.13.0-0.0.pre.103 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 509ddfda5c (7 hours ago), 2022-03-15 21:10:18 -0700
    • Engine revision dc22c4c7fe
    • Dart version 2.17.0 (build 2.17.0-202.0.dev)
    • DevTools version 2.11.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that
        responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.2.1 21D62 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

--- a/packages/flutter/lib/fix_data.yaml
+++ b/packages/flutter/lib/fix_data.yaml
@@ -17,6 +17,20 @@
 
 version: 1
 transforms:
+  # Changes made in https://github.com/flutter/flutter/pull/100381
+  - title: "Migrate 'TextSelectionOverlay.fadeDuration' to SelectionOverlay.fadeDuration"
+    date: 2022-03-18
+    element:
+      uris: [ 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+      field: 'fadeDuration'
+      inClass: 'TextSelectionOverlay'
+    changes:
+      - kind: 'replacedBy'
+        newElement:
+          uris: ['widgets.dart']
+          field: 'fadeDuration'
+          inClass: 'SelectionOverlay'
+
   # Changes made in https://github.com/flutter/flutter/pull/96174
   - title: "Migrate 'useDeleteButtonTooltip' to 'deleteButtonTooltipMessage'"
     date: 2022-01-05

--- a/packages/flutter/lib/fix_data.yaml
+++ b/packages/flutter/lib/fix_data.yaml
@@ -27,7 +27,7 @@ transforms:
     changes:
       - kind: 'replacedBy'
         newElement:
-          uris: ['widgets.dart']
+          uris: [ 'widgets.dart', 'material.dart', 'cupertino.dart' ]
           field: 'fadeDuration'
           inClass: 'SelectionOverlay'
 

--- a/packages/flutter/lib/src/widgets/text_selection.dart
+++ b/packages/flutter/lib/src/widgets/text_selection.dart
@@ -303,6 +303,13 @@ class TextSelectionOverlay {
     );
   }
 
+  /// Controls the fade-in and fade-out animations for the toolbar and handles.
+  @Deprecated(
+    'Use `SelectionOverlay.fadeDuration` instead. '
+    'This feature was deprecated after 2.12.0-4.1.pre.'
+  )
+  static const Duration fadeDuration = SelectionOverlay.fadeDuration;
+
   // TODO(mpcomplete): what if the renderObject is removed or replaced, or
   // moves? Not sure what cases I need to handle, or how to handle them.
   /// The editable line in which the selected text is being displayed.

--- a/packages/flutter/lib/src/widgets/text_selection.dart
+++ b/packages/flutter/lib/src/widgets/text_selection.dart
@@ -306,7 +306,7 @@ class TextSelectionOverlay {
   /// Controls the fade-in and fade-out animations for the toolbar and handles.
   @Deprecated(
     'Use `SelectionOverlay.fadeDuration` instead. '
-    'This feature was deprecated after 2.12.0-4.1.pre.'
+    'This feature was deprecated after v2.12.0-4.1.pre.'
   )
   static const Duration fadeDuration = SelectionOverlay.fadeDuration;
 

--- a/packages/flutter/test/widgets/text_selection_test.dart
+++ b/packages/flutter/test/widgets/text_selection_test.dart
@@ -91,6 +91,10 @@ void main() {
     );
   }
 
+  test('TextSelectionOverlay.fadeDuration exist', () async {
+    expect(TextSelectionOverlay.fadeDuration, SelectionOverlay.fadeDuration);
+  });
+
   testWidgets('a series of taps all call onTaps', (WidgetTester tester) async {
     await pumpGestureDetector(tester);
     await tester.tapAt(const Offset(200, 200));

--- a/packages/flutter/test_fixes/cupertino.dart
+++ b/packages/flutter/test_fixes/cupertino.dart
@@ -226,4 +226,7 @@ void main() {
   bool nowShowing = scrollbar.isAlwaysShown;
   RawScrollbar rawScrollbar = RawScrollbar(isAlwaysShown: true);
   nowShowing = rawScrollbar.isAlwaysShown;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  TextSelectionOverlay.fadeDuration;
 }

--- a/packages/flutter/test_fixes/cupertino.dart.expect
+++ b/packages/flutter/test_fixes/cupertino.dart.expect
@@ -226,4 +226,7 @@ void main() {
   bool nowShowing = scrollbar.thumbVisibility;
   RawScrollbar rawScrollbar = RawScrollbar(thumbVisibility: true);
   nowShowing = rawScrollbar.thumbVisibility;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  SelectionOverlay.fadeDuration;
 }

--- a/packages/flutter/test_fixes/material.dart
+++ b/packages/flutter/test_fixes/material.dart
@@ -551,4 +551,7 @@ void main() {
   rawChip = RawChip(useDeleteButtonTooltip: true);
   rawChip = RawChip(useDeleteButtonTooltip: false, deleteButtonTooltipMessage: 'Delete Tooltip');
   rawChip.useDeleteButtonTooltip;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  TextSelectionOverlay.fadeDuration;
 }

--- a/packages/flutter/test_fixes/material.dart.expect
+++ b/packages/flutter/test_fixes/material.dart.expect
@@ -524,4 +524,7 @@ void main() {
   rawChip = RawChip();
   rawChip = RawChip(deleteButtonTooltipMessage: 'Delete Tooltip');
   rawChip.deleteButtonTooltipMessage;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  SelectionOverlay.fadeDuration;
 }

--- a/packages/flutter/test_fixes/widgets.dart
+++ b/packages/flutter/test_fixes/widgets.dart
@@ -181,4 +181,7 @@ void main() {
   // Changes made in https://github.com/flutter/flutter/pull/96957
   RawScrollbar rawScrollbar = RawScrollbar(isAlwaysShown: true);
   nowShowing = rawScrollbar.isAlwaysShown;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  TextSelectionOverlay.fadeDuration;
 }

--- a/packages/flutter/test_fixes/widgets.dart.expect
+++ b/packages/flutter/test_fixes/widgets.dart.expect
@@ -181,4 +181,7 @@ void main() {
   // Changes made in https://github.com/flutter/flutter/pull/96957
   RawScrollbar rawScrollbar = RawScrollbar(thumbVisibility: true);
   nowShowing = rawScrollbar.thumbVisibility;
+
+  // Change made in https://github.com/flutter/flutter/pull/100381
+  SelectionOverlay.fadeDuration;
 }

